PureBASIC Lang Def v1.4

tajmone opened the pull request:
BUG-FIX: Added sanitize function to avoid false positive Escape Sequences in structured vars (eg: "\v" in "SomeStructure\var1").

Thanks to André Simon -- see Issue #23:
-- https://github.com/andre-simon/highlight/issues/23

tajmone committed:
PureBASIC Lang Def v1.4

BUG-FIX: Added sanitize function to avoid false positive Escape Sequences in structured vars (eg: "\v" in "SomeStructure\var1").

Thanks to André Simon -- see Issue #23:
-- https://github.com/andre-simon/highlight/issues/23

tajmone committed:
Lang Def v1.5

IMPROVEMENTS:
- Escaped-String Prefix (~) is no longer handled as a keyword (ID=4/kwd) but its recognized as a valid string delimiter.
- Escape Sequences are further sanitized so that they can occur only inside strings with the ~" opening delimiter.

ABROGATED:
- The lang definition no longer uses Keyword 4 (Escaped-String Prefix).

andre-simon commented:
See my comment about another fix: https://github.com/andre-simon/highlight/issues/23

tajmone commented:
Please don't merge these 2 commits yet ... I need to fix some issues which popped up in some edge-cases.

Hopefully, in a few days I'll fix them, carry some thorough testing for all edge cases and push another commit here.

tajmone committed:
Lang Def v1.6

FIXES: Strings parsing.
ADDS: Numbers definitions (Hex, Binary, Floats, Decimals)
REMOVES: Escape sequences (suppressed due to some issues that break strings)

tajmone commented:
The last commit (`d87b2e`, lang def v1.6) finally fixes the strings problems at the expense of escaped sequences (temporarily suppressed, until I find a solution and test it throughly).

It also adds numbers definitions.

So this is good to merge (you can squash it a single commit if you prefer).

tajmone committed:
Lang Def v1.7

IMPROVEMENTS: Now implements escapes sequences inside strings. Thoroughly tested.

tajmone commented:
Finally I've manged (`639651e`) to implement capturing escape sequences inside strings. It took a lot of testing to catch all edge-cases were strings could get corrupted, but at the end I've made it!
